source/uuid.c

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+#include <aws/common/uuid.h>
+
+#include <aws/common/byte_buf.h>
+#include <aws/common/device_random.h>
+
+#include <inttypes.h>
+#include <stdio.h>
+
+#define UUID_FORMAT                                                                                                    \
+    "%02" SCNx8 "%02" SCNx8 "%02" SCNx8 "%02" SCNx8 "-%02" SCNx8 "%02" SCNx8 "-%02" SCNx8 "%02" SCNx8 "-%02" SCNx8     \
+    "%02" SCNx8 "-%02" SCNx8 "%02" SCNx8 "%02" SCNx8 "%02" SCNx8 "%02" SCNx8 "%02" SCNx8
+
+#include <stdio.h>
+
+#ifdef _MSC_VER
+/* disables warning non const declared initializers for Microsoft compilers */
+#    pragma warning(disable : 4204)
+#    pragma warning(disable : 4706)
+/* sprintf warning (we already check the bounds in this case). */
+#    pragma warning(disable : 4996)
+#endif
+
+int aws_uuid_init(struct aws_uuid *uuid) {
+    struct aws_byte_buf buf = {
+        .buffer = uuid->uuid_data,
+        .len = 0,
+        .capacity = sizeof(uuid->uuid_data),
+        .allocator = NULL,
+    };
+    return aws_device_random_buffer(&buf);
```

Comment:
using urandom isn't actually going to be globally unique without a machine id (MAC addr or similar, MACs are lies in the cloud) + timestamp + monotonic counter. Do we care?

See: https://github.com/aws/lumberyard/blob/master/dev/Code/Framework/AzCore/AzCore/Component/Entity.cpp#L1420 for a possibly imperfect, but so far never collided implementation :)

Reply:
The probability of a clash from urandom should be lower than a clash from cloned Mac addresses with a matching time stamp.